catalog/services_state.go

```diff
@@ -354,11 +355,25 @@ func (state *ServicesState) Format(list *memberlist.Memberlist) string {
 
 	refTime := time.Now().UTC()
 
+	var servers []*Server
+	for _, svr := range state.Servers {
```

Comment:
Note sure where this gets used besides the Print method below, but I'm wondering if it needs to lock the state.

Reply:
It is used in `Print` and also in the  http `/servers` endpoint. Also for debugging. It doesn't lock the state... calling routines are expected to do that. It may get called inside a lock.